app/views.py

```diff
@@ -0,0 +1,131 @@
+from flask_security.utils import hash_password
+from datetime import datetime, timedelta
+
+
+@app.before_first_request
+def before_first_request():
+    """
+    runs before the first request, adds an user account and sets it as 'admin' role.
+    """
+
+    # create admin and normal user roles
+    user_datastore.find_or_create_role(name='admin', description='Administrator')
+    user_datastore.find_or_create_role(name='user', description='End user')
+
+    # create an admin user and add to database
+    encrypted_password = hash_password('password')
+    if not user_datastore.get_user('admin@mporter.co'):
+        user_datastore.create_user(email='admin@mporter.co', password=encrypted_password)
+
+    db_init.session.commit()
+
+    # make admin@mporter.co the admin user
+    user_datastore.add_role_to_user('admin@mporter.co', 'admin')
+    db_init.session.commit()
+
+
+@app.route('/')
+def index():
+    """
+    serves the static index page with features and stuff
+    """
+    return render_template('index.html')
+
+
+@app.route('/mentee')
+@login_required
+def mentee():
+    """
+    serves a page with tasks by the mentee and a form to add new tasks
+    additionally a button to logout. KISS.
+    """
+    current_user_id = current_user.get_id()
+    user = user_datastore.get_user(current_user_id)
+
+    is_admin = False
+    if len(user.roles) > 0:
+        is_admin = True if user.roles[0].name == 'admin' else False
+
+    from .models import Tasks, Mentees
+
+    user_tasks = Tasks\
+        .query\
+        .filter_by(mentee_id=user.id) \
+        .filter(Tasks.at_created >= datetime.today() - timedelta(days=1))\
+        .all()
+
+    mentee_obj = Mentees.query.filter_by(id=user.id).first()
+
+    user_obj = {
+        'user_id': user.id,
+        'user_email': user.email,
+        'user_tasks': user_tasks,
+        'user_mentors': mentee_obj.mentor,
+        'is_admin': is_admin
+    }
+
+    return render_template('mentee.html', user=user_obj)
+
+
+@app.route('/new-task', methods=['POST'])
+@login_required
+def new_task():
+    """
+    An api to add a new task. API needs authentication, and the task will be added to the authenticated user's tasks.
+    @:param task: text of the task to be added in post request body
+    """
+    task = request.form.get('task')
+    current_user_id = current_user.get_id()
+    user = user_datastore.get_user(current_user_id)
+
+    from .models import Tasks
+
+    task = Tasks(mentee_id=user.id, task=task)
+
+    db_init.session.add(task)
+    db_init.session.commit()
+
+    return redirect(url_for('mentee'))
+
+
+@app.route('/new-mentor', methods=['POST'])
+@login_required
+def new_mentor():
+    """
+    add a mentor to user's mentors list. If mentor is not present, add a new mentor as well with the name supplied
+    note that if a mentor is present, the mentor_name supplied by user is ignored
+    @:param: mentor-name, mentor-email
+    """
+    from .models import Mentors, Mentees
+
```

Comment:
[Discussion] Use  an MVCS pattern? https://stackoverflow.com/questions/5702391/mvcs-model-view-controller-service

Reply:
https://github.com/webappsolution/flow-mvc/wiki/MVCS-Architecture

Reply:
Interesting. Yeah, I intuitively felt I'm cluttering both the models and the views. Will read up

Reply:
i really want to have this discussion. AFAIR you had mentioned that usually this pattern is followed in Frontend development

Reply:
You're right. I did tell you, and in angular, they're a huge thing (https://docs.angularjs.org/guide/services). Even in react, we had a services layer in our lg codebase. Not sure what should form a service and what not. I mean, what logic can be written on the model level, and what not.

Another problem is circular dependencies. Is it a standard problem with these python apps or am I messing up some very basics?